docs/key-concepts.md

```diff
@@ -189,3 +189,43 @@ tree, and {func}`jax.tree.reduce` can be used to apply a reduction across the le
 in a tree.
 
 You can learn more in the {ref}`working-with-pytrees` tutorial.
+
+(key-concepts-prngs)=
+## Pseudorandom numbers
+
+Generally, JAX strives to be compatible with NumPy, but pseudo random number generation is a notable exception. In NumPy, pseudo random number generation is based on a global `state`, which can be set using {func}`numpy.random.seed`. Global random state interacts poorly with JAX's compute model and makes it difficult to enforce reproducibility across different threads, processes, and devices. JAX instead tracks state explicitly via a random `key`:
```

Comment:
Maybe not for this PR, but just to put a pin in this: a while back one of the NumPy maintainers complained to me that this is misleading: NumPy doesn't recommend using a global state, but rather passing around a local state based on `np.random.default_prng()`. We should probably rework this discussion to avoid the appearance of an unfair comparison.

Reply:
The issue then becomes less about "implicit vs. explicit state", but rather one of "implicit vs. explicit updates of the explicit state"

Reply:
That makes sense. I softened this a little to not imply that NumPy requires using global state.

Reply:
I'll file an issue to rework it further.